src/utils/program-enrolment-helper.ts

```diff
@@ -0,0 +1,71 @@
+export function isPostSubmissionEnabled(expression: string, encounters: any[]): boolean {
```

Comment:
this function is doing more than checking if a `postSubmissionAction` is enabled... please rename this and also check if there is no function in the form engine that already does this... how `hideWhenExpression` is processed ?

Reply:
There's an existing `expression-runner` but after studying it, I realised it couldn't be reused for this scenario:
The expression runner has access to all fields so expressions are evaluated in relation to all dependent fields regardless of whether they have data or not. The `post submission action` on the other hand has no access to all fields but has access to the submitted encounter and the submitted encounter eliminates fields/obs that don't have values, so it's just a few fields that get submitted.
The expression-runner also uses the `encounter context` which is not present at post-submission.

Reply:
@kajambiya but the name of this function is not inline with what it does.